projects/plugins/crm/src/automation/interface-step.php

```diff
@@ -36,13 +36,40 @@ public function execute( Data_Type $data );
 	public function get_next_step_id();
 
 	/**
-	 * Set the next step.
+	 * Get the next step if the current one is successful.
 	 *
 	 * @since $$next-version$$
 	 *
-	 * @param int|string|null $step_id The next linked step.
+	 * @return int|string|null The next linked step id.
 	 */
-	public function set_next_step( $step_id );
+	public function get_next_step_true();
+
+	/**
+	 * Set the next step if the current one is successful.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return int|string|null The next linked step id.
+	 */
+	public function set_next_step_true();
```

Comment:
Do we need this?

Reply:
Due to the way we've implemented the system, then yes.

Because we use `<something> instanceof Step` then we don't enforce folks to extend our abstract base class (`Base_step`), so 3. party developers could technically avoid the base class right now, so we would force them.

I would personally prefer if we had consistently used `Base_Step` and `is_subclass_of( <something>, Base_Step::class )` so we could assume everything that exists in the abstract base class would be present, but we haven't